subsys/app_event_manager/app_event_manager_shell.c

```diff
@@ -166,3 +167,27 @@ SHELL_STATIC_SUBCMD_SET_CREATE(sub_app_event_manager,
 
 SHELL_CMD_REGISTER(app_event_manager, &sub_app_event_manager,
 		   "Application Event Manager commands", NULL);
+
+static int button_sim_handler(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 3) {
+		shell_error(shell, "Missing required parameter(s)\n");
+		return -EINVAL;
+	}
+
+	bool pressed = false;
+	struct button_event *event = new_button_event();
+
+	if (strcmp(argv[2], "true") == 0) {
+		pressed = true;
+	}
+
+	event->key_id = (uint16_t)argv[1];
+	event->pressed = pressed;
+	APP_EVENT_SUBMIT(event);
+
+	return 0;
+}
+
+SHELL_CMD_REGISTER(button_sim, NULL,
```

Comment:
Can you integrate into the `SHELL_STATIC_SUBCMD_SET_CREATE` call above instead of creating a totally separate command?